mchess-master/mchess-master/CBoard.cc

```diff
@@ -241,55 +241,50 @@ void CBoard::find_legal_moves(CMoveList &moves) const
 {
     moves.clear();
 
-    if (m_side_to_move > 0)
+    if (m_side_to_move > 0){
+        int iscaptured=0;
+        int prerun=0;
         for (int i=A1; i<=H8; ++i)
         {
             int8_t piece = m_board[i];
             int j;
-
+            if(iscaptured==1&&prerun==0)
+                moves.clear();
```

Comment:
So Advise you create 2 stacks, 1 called _**capture_moves**_, one called _**non_capture_moves**_. Push all moves into _**capture_moves**_, and only push into _**non_capture_moves**_ when there are no captures available. This will significantly save memory, since you don't need to check for non_captures when you don't have to